cmd/clusterctl/client/cluster/cert_manager.go

```diff
@@ -89,21 +81,68 @@ type CertManagerClient interface {
 
 // certManagerClient implements CertManagerClient .
 type certManagerClient struct {
-	configClient        config.Client
-	proxy               Proxy
-	pollImmediateWaiter PollImmediateWaiter
+	configClient                       config.Client
+	proxy                              Proxy
+	pollImmediateWaiter                PollImmediateWaiter
+	embeddedCertManagerManifestVersion string
+	embeddedCertManagerManifestHash    string
 }
 
 // Ensure certManagerClient implements the CertManagerClient interface.
 var _ CertManagerClient = &certManagerClient{}
 
+func (cm *certManagerClient) getManifestHash() (string, error) {
+	yamlData, err := manifests.Asset(embeddedCertManagerManifestPath)
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("%x", sha256.Sum256(yamlData)), nil
+}
+
+func (cm *certManagerClient) getManifestVersion() error {
+	// Gets the cert-manager objects from the embedded assets.
+	objs, err := cm.getManifestObjs()
+	if err != nil {
+		return err
+	}
+	found := false
+
+	for i := range objs {
+		o := objs[i]
+		if o.GetKind() == "CustomResourceDefinition" {
+			labels := o.GetLabels()
+			version, ok := labels[certmanagerVersionLabel]
+			if ok {
+				s := strings.Split(version, "-")
+				cm.embeddedCertManagerManifestVersion = s[2]
+				found = true
+				break
+			}
+		}
+	}
+	if !found {
+		return errors.Errorf("Failed to found Label %s in all CRDs", certmanagerVersionLabel)
+	}
+	return nil
+}
+
 // newCertMangerClient returns a certManagerClient.
 func newCertMangerClient(configClient config.Client, proxy Proxy, pollImmediateWaiter PollImmediateWaiter) *certManagerClient {
-	return &certManagerClient{
-		configClient:        configClient,
-		proxy:               proxy,
-		pollImmediateWaiter: pollImmediateWaiter,
+	yaml, _ := manifests.Asset(embeddedCertManagerManifestPath)
+	hash := fmt.Sprintf("%x", sha256.Sum256(yaml))
```

Comment:
should be achieved by calling getManifestHash, so we have this computation implemented only in one place?

Reply:
yes, thanks for the reminder, I also find it just now